core/src/main/scala/org/apache/spark/input/WholeTextFileRecordReader.scala

```diff
@@ -108,8 +108,17 @@ private[spark] class ConfigurableCombineFileRecordReader[K, V](
   override def initNextRecordReader(): Boolean = {
     val r = super.initNextRecordReader()
     if (r) {
-      this.curReader.asInstanceOf[HConfigurable].setConf(getConf)
+      if (getConf != null) {
+        this.curReader.asInstanceOf[HConfigurable].setConf(getConf)
```

Comment:
This is needed because `initNextRecordReader` could be called in the constructor, which `getConf` would be null.

We have to override `setConf` too to set conf for the first reader.